tests/core/arch-arm/armrelocator.c

```diff
@@ -260,13 +264,31 @@ branch_scenario_execute (BranchScenario * bs,
     calculated_lr = (guint32) (fixture->aw.pc +
         (bs->expected_lr_distance * sizeof (guint32)));
     *((guint32 *) (bs->expected_output + bs->lr_offset)) =
+#if G_BYTE_ORDER == G_LITTLE_ENDIAN
         GUINT32_TO_LE (calculated_lr);
+#else
+        GUINT32_TO_BE (calculated_lr);
+#endif
   }
 
   g_assert_cmpuint (gum_arm_relocator_read_one (&fixture->rl, &insn), ==, 4);
   g_assert_cmpint (insn->id, ==, bs->instruction_id);
   g_assert_true (gum_arm_relocator_write_one (&fixture->rl));
   gum_arm_writer_flush (&fixture->aw);
-  g_assert_cmpint (memcmp (fixture->output, bs->expected_output,
-      bs->expected_output_length * sizeof (guint32)), ==, 0);
+  gboolean same_content = memcmp (fixture->output, bs->expected_output,
+      bs->expected_output_length * sizeof (guint32)) == 0;
+
+  gchar * diff = test_util_diff_binary (
```

Comment:
Declaration after statement. Variables should be declared at the top of the nearest scope, in the order that they're used in. (Except for the value used to hold the return value, which should be first.)